packages/loader/container-loader/src/loader.ts

```diff
@@ -428,3 +493,52 @@ export class Loader implements IHostLoader {
 		);
 	}
 }
+
+/**
+ * {@inheritDoc @fluidframework/container-definitions#ILoader.resolve}
+ * @legacy
+ * @alpha
+ */
+export async function resolveContainer(
+	loaderResolveContainerProps: IResolveContainerProps,
+): Promise<IContainer> {
+	const loader = new Loader(loaderResolveContainerProps);
+	return loader.resolve(
+		loaderResolveContainerProps.request,
+		loaderResolveContainerProps.pendingLocalState,
+	);
+}
+
+/**
+ * Creates a new container using the specified code details but in an unattached state. While unattached, all
+ * updates will only be local until the user explicitly attaches the container to a service provider.
+ * @param createDetachedContainerProps - Services and properties necessary for creating detached container.
+ * @legacy
+ * @alpha
+ */
+export async function createDetachedContainer(
+	createDetachedContainerProps: ICreateDetachedContainerProps,
+): Promise<IContainer> {
+	const loader = new Loader(createDetachedContainerProps);
```

Comment:
Not for this PR necessarily (unless it is helpful), but I imagine that `Container`'s static methods (`Container.createDetached`, `Container.load`, etc.) should also go away and might help clarify the props typing story - the current division of labor between the loader class, the static methods, the container constructor, the private `container.load` is unintuitive at best.